fix(federation): differentiate field and input value names (#280)
apollographql/federation: federation-js/CHANGELOG.md

> The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.

> - Fix check identical types when having fields and arguments wit the same name [PR #280](https://github.com/apollographql/federation/pull/280)

Reviewer: ```suggestion
- Fix check for value types when having fields and arguments with the same name [PR #280](https://github.com/apollographql/federation/pull/280)
```
